### PR TITLE
refactor(cdk): deprecate RX_ANGULAR_CONFIG 

### PR DESCRIPTION
**Pull request by BioPhoton**
Deprecate global config in favour of `RX_ANGULAR_RENDER_STRATEGIES_CONFIG`

### CODE REVIEW

File: apps/demos/src/app/app.module.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ import { HomeComponent } from './features/home/home.component';
       useValue: environment,
     },
     {
-      provide: RX_ANGULAR_CONFIG,
+      provide: RX_ANGULAR_RENDER_STRATEGIES_CONFIG,
```
Comment (hoebbelsB) [inline apps/demos/src/app/app.module.ts:25]: we could prefix such consts with just `RXA_` which would decrease the length of the name quite a bit: `RXA_RENDER_STRATEGIES_CONFIG`.

We could also think of not postfixing it with CONFIG => `RXA_RENDER_STRATEGIES`

what do u say?

File: libs/cdk/render-strategies/src/lib/config.ts [hunk 2]
```diff
@@ -3,37 +3,66 @@ import { RX_CONCURRENT_STRATEGIES } from './concurrent-strategies';
 import { RxCustomStrategyCredentials, RxDefaultStrategyNames, RxStrategyNames } from './model';
 import { RX_NATIVE_STRATEGIES } from './native-strategies';
```
Comment (hoebbelsB) [inline libs/cdk/render-strategies/src/lib/config.ts:4]: this one is actually prefixed with just `RX_` ...

File: libs/cdk/render-strategies/src/lib/strategy-provider.service.ts [hunk 3]
```diff
@@ -126,15 +131,20 @@ export class RxStrategyProvider<T extends string = string> {
...
+   */
   constructor(
     @Optional()
     @Inject(RX_ANGULAR_CONFIG)
-    cfg: RxAngularConfig<T>
+    @Optional()
+    cfgOld: RxRenderStrategiesConfig<T>,
+    @Optional()
+    @Inject(RX_ANGULAR_RENDER_STRATEGIES_CONFIG)
+    cfg: RxRenderStrategiesConfig<T>
```
Comment (hoebbelsB) [inline libs/cdk/render-strategies/src/lib/strategy-provider.service.ts:144]: one of those `@Optional` is too much :D

**Review by hoebbelsB — CHANGES_REQUESTED** [review]
looks good <3 , just some comments about naming

File: libs/cdk/render-strategies/src/lib/strategy-provider.service.ts (hunk 3 above)
Comment (BioPhoton) [reply libs/cdk/render-strategies/src/lib/strategy-provider.service.ts]: thx

**Pull request merged**